src/apis/impl/NotificationApiImpl.ts

```diff
@@ -97,4 +94,27 @@ export class NotificationApiImpl implements NotificationApi {
     });
   }
 
+  async concatenateFilterResults(filter: Filter<Notification>, nextId?: string | undefined, limit?: number | undefined, accumulator: Array<Notification> = []): Promise<Array<Notification>> {
+    const paginatedNotifications = await this.filterNotifications(filter, nextId, limit);
+    return !paginatedNotifications.nextKeyPair?.startKeyDocId
+      ? accumulator.concat(paginatedNotifications.rows)
+      : this.concatenateFilterResults(filter, paginatedNotifications.nextKeyPair.startKeyDocId, limit, accumulator.concat(paginatedNotifications.rows))
+  }
+
+  async getPendingNotificationsFromNewUsers(): Promise<Array<Notification>> {
+    const user = await this.userApi.getCurrentUser();
+    if (!user) throw new Error("There is no user currently logged in");
+    if (!user.healthcarePartyId) throw new Error("User is not a Healthcare Party");
```

Comment:
Check if the user is a dataOwner instead of a HCP : A patient can also receive notifications